js-api-doc/value/color.d.ts

```diff
@@ -12,117 +90,456 @@ export class SassColor extends Value {
+   *
+   * @throws `Error` if `channel` is not `alpha` or a channel in `space`.
+   */
+  channel(
+    channel: ChannelNameHSL | 'alpha',
+    options: {space: ColorSpaceHSL}
+  ): number;
+  channel(
+    channel: ChannelNameHWB | 'alpha',
+    options: {space: ColorSpaceHWB}
+  ): number;
+  channel(
+    channel: ChannelNameLab | 'alpha',
+    options: {space: ColorSpaceLab}
+  ): number;
+  channel(
+    channel: ChannelNameLCH | 'alpha',
+    options: {space: ColorSpaceLCH}
+  ): number;
+  channel(
+    channel: ChannelNameRGB | 'alpha',
+    options: {space: ColorSpaceRGB}
+  ): number;
+  channel(
+    channel: ChannelNameXYZ | 'alpha',
+    options: {space: ColorSpaceXYZ}
+  ): number;
 
   /** This color's alpha channel, between `0` and `1`. */
   get alpha(): number;
 
   /**
-   * Changes one or more of this color's RGB channels and returns the result.
+   * Returns a boolean indicating whether a given channel value is a [missing
+   * channel].
+   *
+   * [missing channel]: https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#missing_color_components
+   */
+  isChannelMissing(channel: ChannelName | 'alpha'): boolean;
+
+  /**
+   * A boolean indicating whether this color's alpha channel is a [missing
+   * channel].
+   *
+   * [missing channel]: https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#missing_color_components
+   */
+  get isAlphaMissing(): boolean;
+
+  /**
+   * Returns a boolean indicating whether a given `channel` is "powerless" in
+   * this color. This is a special state that's defined for individual color
+   * spaces, which indicates that a channel's value won't affect how a color is
+   * displayed.
+   */
+  isChannelPowerless(channel: ChannelName): boolean;
+  isChannelPowerless(
+    channel: ChannelNameHSL,
+    options?: {space: ColorSpaceHSL}
+  ): boolean;
+  isChannelPowerless(
+    channel: ChannelNameHWB,
+    options?: {space: ColorSpaceHWB}
+  ): boolean;
+  isChannelPowerless(
+    channel: ChannelNameLab,
+    options?: {space: ColorSpaceLab}
+  ): boolean;
+  isChannelPowerless(
+    channel: ChannelNameLCH,
+    options?: {space: ColorSpaceLCH}
+  ): boolean;
+  isChannelPowerless(
+    channel: ChannelNameRGB,
+    options?: {space: ColorSpaceRGB}
+  ): boolean;
+  isChannelPowerless(
+    channel: ChannelNameXYZ,
+    options?: {space: ColorSpaceXYZ}
+  ): boolean;
+
+  /**
+   * Returns a color partway between this color and passed `color2`.
    */
-  change(options: {
-    red?: number;
-    green?: number;
-    blue?: number;
-    alpha?: number;
+  interpolate(options: {color2: SassColor; weight?: number}): SassColor;
+  interpolate(options: {
+    color2: SassColor;
+    weight?: number;
+    method?: HueInterpolationMethod;
   }): SassColor;
 
   /**
    * Changes one or more of this color's HSL channels and returns the result.
+   *
+   * @throws `Error` if `space` is missing and this color is not a legacy color
+   * space (`rgb`, `hsl`, or `hwb`).
```

Comment:
A color can't be a color space. It could be a legacy color, or it could be _in_ a legacy color space.